src/charm.py

```diff
@@ -71,7 +81,17 @@ def __init__(self, *args):
 
         self.framework.observe(self.on.agent_pebble_ready, self.on_pebble_ready)
         self.framework.observe(
-            self.on["prometheus-remote-write"].relation_changed, self.on_remote_write_changed
+            self.on[REMOTE_WRITE_RELATION_NAME].relation_changed, self.on_remote_write_changed
+        )
+        self.framework.observe(
+            self.on[REMOTE_WRITE_RELATION_NAME].relation_joined, self.update_metrics_rules
+        )
+        self.framework.observe(self.on.upgrade_charm, self.update_metrics_rules)
+        self.framework.observe(
```

Comment:
Why not using the https://github.com/canonical/prometheus-k8s-operator/blob/e964e639274efff359de78b65c261c9a9cf981be/lib/charms/prometheus_k8s/v0/prometheus_scrape.py#L215 event?

Reply:
Yes, please.